### PR TITLE
Qt add Terms & Conditions

### PR DESCRIPTION
**Pull request by ghost**
Add T&C

### CODE REVIEW

File: selfdrive/version.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def get_git_remote(default: Optional[str] = None) -> Optional[str]:
 prebuilt = os.path.exists(os.path.join(BASEDIR, 'prebuilt'))
 
 training_version: bytes = b"0.2.0"
-terms_version: bytes = b"2"
+terms_version: bytes = b"3"
```
Comment (pd0wm) [inline selfdrive/version.py:48]: For testing? Remove so we don't accidentally merge it in.

Comment (grekiki) [reply selfdrive/version.py:48]: Well it is a new version :). Ok I'll remove it

File: selfdrive/ui/qt/offroad/onboarding.cc [hunk 2]
```diff
@@ -121,7 +121,16 @@ void OnboardingWindow::updateActiveScreen() {
   }
 }
 
+void enableBacklight(){
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/onboarding.cc:124]: this shouldn't go in onboarding. ideally we want a nice HW abstraction in cpp like we have in python

Comment (adeebshihadeh) [reply selfdrive/ui/qt/offroad/onboarding.cc:124]: it should also be gated on QCOM2

Comment (pd0wm) [reply selfdrive/ui/qt/offroad/onboarding.cc:124]: It's also only a problem for testing when you keep restarting the ui. On boot the backlight is already set to max.

File: selfdrive/ui/qt/offroad/onboarding.cc [hunk 3]
```diff
@@ -67,31 +72,26 @@ TrainingGuide::TrainingGuide(QWidget* parent) {
...
-    font-size: 75px;
-    border-radius: 10px;
-    background-color: #292929;
-  )");
-  main_layout->addWidget(terms, 1, 0, 1, -1);
-  main_layout->setRowStretch(1, 1);
-
+  QVBoxLayout *main_layout = new QVBoxLayout;
+  main_layout->setContentsMargins(40, 0, 40, 0);
+  QWebEngineView *view = new QWebEngineView(this);
```
Comment (pd0wm) [inline selfdrive/ui/qt/offroad/onboarding.cc:77]: Think this slows ui start a lot. We should only load chromium when the terms actually have to be displayed.

Comment (pd0wm) [reply selfdrive/ui/qt/offroad/onboarding.cc:77]: Just make sure the widget is never created if we're not going to show it.

Comment (grekiki) [reply selfdrive/ui/qt/offroad/onboarding.cc:77]: Fixed

File: selfdrive/ui/qt/offroad/onboarding.hpp [hunk 4]
```diff
@@ -37,10 +40,15 @@ class OnboardingWindow : public QStackedWidget {
...
+  QPushButton* accept_btn;
 
 signals:
   void onboardingDone();
 
 public slots:
   void updateActiveScreen();
+
+private slots:
+  void scrollPosition(QPointF position);
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/onboarding.hpp:53]: can we give this a more descriptive name?

Comment (pd0wm) [inline selfdrive/ui/qt/offroad/onboarding.hpp:53]:
```suggestion
  void scrollPositionChanged(QPointF position);
```

File: selfdrive/ui/qt/offroad/onboarding.cc [hunk 5]
```diff
@@ -145,10 +154,21 @@ OnboardingWindow::OnboardingWindow(QWidget *parent) : QStackedWidget(parent) {
...
+    QPushButton:disabled {
+      color: #777777;
+      background-color: #222222;
+    }
   )");
 
   updateActiveScreen();
 }
+
+void OnboardingWindow::scrollPosition(QPointF position){
```
Comment (pd0wm) [inline selfdrive/ui/qt/offroad/onboarding.cc:169]:
```suggestion
void OnboardingWindow::scrollPositionChanged(QPointF position){
```

File: selfdrive/ui/qt/offroad/onboarding.cc [hunk 6]
```diff
@@ -70,31 +73,31 @@ TrainingGuide::TrainingGuide(QWidget* parent) {
...
+  accept_btn = new QPushButton("Scroll to accept");
+  accept_btn->setEnabled(false);
+  buttons->addWidget(accept_btn);
   QObject::connect(accept_btn, &QPushButton::released, [=]() {
     Params().write_db_value("HasAcceptedTerms", current_terms_version);
     updateActiveScreen();
   });
 
-  main_layout->addWidget(new QPushButton("Decline"), 2, 0);
+  QObject::connect(view->page(), SIGNAL(scrollPositionChanged(QPointF)), this, SLOT(scrollPosition(QPointF)));
```
Comment (pd0wm) [inline selfdrive/ui/qt/offroad/onboarding.cc:96]:
```suggestion
  QObject::connect(view->page(), SIGNAL(scrollPositionChanged(QPointF)), this, SLOT(scrollPositionChanged(QPointF)));
```

**Pull request merged**